astropy/units/tests/test_interaction_with_others.py

```diff
@@ -0,0 +1,66 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Test that astropy's Unit class can deal with units from other projects."""
+
+import pytest
+
+from astropy.units import Unit, UnitBase, dimensionless_unscaled
+
+
+pytestmark = pytest.mark.filterwarnings("ignore")
+
+
+class OtherUnitModuleTests:
+    # Subclasses must define a setup_class that sets
+    # - oum for the module to the units.
+    # - dimensionless to its dimensionless unit.
+
+    @pytest.mark.parametrize('unit', ['m', 's', 'kg', 'g', 'cm'])
+    def test_simple(self, unit):
+        u_unit = getattr(self.oum, unit)
+        assert str(u_unit == unit)
+        u_m = Unit(u_unit)
+        assert u_m == Unit(unit)
+
+    def test_dimensionless(self):
+        u_dimensionless = Unit(self.dimensionless)
+        assert isinstance(u_dimensionless, UnitBase)
+        assert u_dimensionless == dimensionless_unscaled
+
+    def test_more_complicated(self):
+        u_unit = self.oum.m / self.oum.s
+        u_m_per_s = Unit(u_unit)
+        assert isinstance(u_m_per_s, UnitBase)
+        assert u_m_per_s == Unit('m/s')
+
+
+class TestUnyt(OtherUnitModuleTests):
+    def setup_class(cls):
+        try:
+            import unyt
+        except ImportError:
+            pytest.skip('unyt not available.')
+
+        cls.oum = unyt
+        cls.dimensionless = unyt.dimensionless
+
+
+@pytest.mark.filterwarnings('ignore')
+class TestAmuse(OtherUnitModuleTests):
+    def setup_class(cls):
+        try:
+            from amuse.units import units as amuse_units
+        except ImportError:
+            pytest.skip('amuse not available.')
+
+        cls.oum = amuse_units
+        cls.dimensionless = amuse_units.none
+
+    def teardown_class(cls):
+        # Annoyingly, amuse thinks it is OK to just override
```

Comment:
@pllim, @bsipocz - this is what I call *really* scary! :ghost:

Reply:
I'm not amused 🤷

Reply:
BTW, found this because numpy raised a deprecation warning - it is good that we are error-ing also on those!

Reply:
I'm more and more concerned of how much maintenance burden this might trigger on the long run as the possible need of following the internals and version dependencies of extra third party packages.

Reply:
Yes, I'm a bit concerned too. Possibly, this should just be in a monthly cron job or so. Alternatively, I could just create amuse and unyt-like classes and check that those work.

Reply:
I'm less concerned about unyt as it's standalone and lightweight and very much in the same pydata ecosystem, but a bit terrified for going into the weeds of amuse

Reply:
I had hoped `amuse-framework` would be the lightweight counterpart -  Maybe @rieder can advice...

Reply:
`amuse-framework` is the AMUSE core plus core libraries that are needed. We're planning (https://github.com/amusecode/amuse/issues/412) to split these up into `amuse-core` (which would be lightweight, python-only) and `amuse-libaries`, but this isn't done yet.